<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>chatbot</class>
 <widget class="QMainWindow" name="chatbot">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1676</width>
    <height>875</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>12</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Gemma Chat Bot -  Your AI Comapnion</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="1,3">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>Chatbot and Chatbot</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1,0,0,0">
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Ice Breaking Sentence</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="txtIceBreaker">
             <property name="text">
              <string>Hi</string>
             </property>
             <property name="placeholderText">
              <string>Write a sentence to start conversation between ChhatBots!</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>in</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinWords">
             <property name="minimum">
              <number>5</number>
             </property>
             <property name="maximum">
              <number>5000</number>
             </property>
             <property name="value">
              <number>60</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Words</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,0,0,1,0">
           <item>
            <widget class="QComboBox" name="cbxModel1">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>40</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnModel1Talk">
             <property name="text">
              <string>Talk</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>40</height>
              </size>
             </property>
             <property name="text">
              <string>with</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cbxModel2">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>40</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnModel2Talk">
             <property name="text">
              <string>Talk</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>User and Chatbot</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1,0">
           <item>
            <widget class="QLineEdit" name="txtChatHistoryPath">
             <property name="placeholderText">
              <string>Chat History File Path</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="cbEnableVoice">
             <property name="text">
              <string>Enable Voice</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,1,0">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Select Large Language Model</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cbxModels">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>40</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnShowDetails">
             <property name="text">
              <string>Show &amp;Details</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0,0">
           <item>
            <widget class="QLineEdit" name="input_area">
             <property name="placeholderText">
              <string>Write your message here!</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnSend">
             <property name="text">
              <string>&amp;Send</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnClearChat">
             <property name="text">
              <string>&amp;Clear Chat</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string>Chatbox</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QTableWidget" name="chat_table">
         <property name="showGrid">
          <bool>false</bool>
         </property>
         <attribute name="horizontalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="horizontalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <column>
          <property name="text">
           <string>chat</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionLoad_Chat_History"/>
   <addaction name="actionSave_Chat_History"/>
   <addaction name="actionSave_As_Chat_History"/>
  </widget>
  <action name="actionSave_Chat_History">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::DocumentSave"/>
   </property>
   <property name="text">
    <string>Save Chat History</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionLoad_Chat_History">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::AppointmentNew"/>
   </property>
   <property name="text">
    <string>Load Chat History</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionSave_As_Chat_History">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::DocumentSaveAs"/>
   </property>
   <property name="text">
    <string>Save As Chat History...</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
